pkg/rsqueue/groups/grouprunner.go

```diff
@@ -41,7 +42,10 @@ func (m *GenericMatcher) Field() string {
 }
 
 func (m *GenericMatcher) Type(workType uint64) (interface{}, error) {
-	return m.types[workType], nil
+	if t, ok := m.types[workType]; ok {
+		return t, nil
+	}
+	return nil, MissingTypeError
```

Comment:
We can wrap this one, too, and then simply pass the error below as suggested above.